avocado/utils/softwareraid.py

```diff
@@ -0,0 +1,161 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2020 IBM
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+# Author: Venkat Rao B <vrbagal1@linux.vnet.ibm.com>
+
+
+import time
+import logging
+from . import process
+
+log = logging.getLogger('avocado.test')
+
+
+class SoftwareRaid:
+    """
+    Perform software raid related operations.
+
+    :param name: Name of the software raid to be created
+    :type name: str
+    :param level: Level of software raid to be created
+    :type name: str
+    :param disk: List of disks for software raid
+    :type disk: list
+    :param metadata: Metadata level for software raid
+    :type metadata: str
+    :param spare: List of spare disks for software raid
+    :type spare: list
+    """
+    def __init__(self, name, level, disk, metadata, spare=''):
```

Comment:
if spare is a list, default should not be a string.

Reply:
Code climate complained it is bad for [] to be a default value. So changed 😀

Reply:
Yes, it is a bad idea to have [] on default values. But you can have = None and change to [] during the code if None.

Reply:
Ah sure, will do that.